lib/tests/streamlit/server_test.py

```diff
@@ -550,6 +550,37 @@ def test_rotates_one_port(
                 )
 
 
+class UnixSocketTest(unittest.TestCase):
+    """Tests start_listening uses a unix socket when socket.address starts with
+       unix:// """
+
+    def get_httpserver(self):
+        httpserver = mock.MagicMock()
+
+        httpserver.add_socket = mock.Mock()
+
+        return httpserver
+
+    def test_unix_socket(self):
+        app = mock.MagicMock()
+
+        os.environ["HOME"] = "/home/koen"
+
+        config.set_option("server.address", "unix://~/fancy-test/testasd")
+        some_socket = object()
+
+        mock_server = self.get_httpserver()
+        with patch.object(
+            tornado.httpserver, "HTTPServer", return_value=mock_server
+        ), patch.object(
+            tornado.netutil, "bind_unix_socket", return_value=some_socket
+        ) as bind_unix_socket:
+            start_listening(app)
+
+            bind_unix_socket.assert_called_with("/home/koen/fancy-test/testasd")
```

Comment:
Same as previous comment -- please change test data to more Obviously Fake test data.